server/src/service/PkiService.js

```diff
@@ -154,6 +158,85 @@ exports.getCurrentCARootCert = async function (envId) {
   };
 };
 
+/**
+ * Returns the Environment Status based on configured information for given envId
+ * envId String ID of environment
+ * returns inline_response_200_1
+ **/
+exports.getEnvStatus = async function (envId) {
+  let envStatus = [
+    {
+      phase: 'BUSINESS_SETUP',
+      steps: [
+        {
+          identifier: 'ID_GENERATION',
+          status: 'NOT_STARTED',
+        }
+      ]
+    },
+    {
+      phase: 'TECNICAL_SETUP',
+      steps: [
+        {
+          identifier: 'ENDPOINTS',
+          status: 'NOT_STARTED',
+        },
+        {
+          identifier: 'CSR_EXCHANGE',
+          status: 'NOT_STARTED'
+        },
+        {
+          identifier: 'CERTIFICATE_AUTHORITY',
+          status: 'NOT_STARTED'
+        },
+        {
+          identifier: 'SERVER_CERTIFICATES_EXCHANGE',
+          status: 'NOT_STARTED'
+        },
+        {
+          identifier: 'JWS_CERTIFICATES',
+          status: 'NOT_STARTED'
+        }
+      ]
+    }
+   ];
+   let dfsp = [];
+   try {
+     dfsp = await DFSPModel.findAllByEnvironment(envId).map(r => dfspRowToObject(r));
+  } catch (error) {
+    if (error instanceof NotFoundError) {
+      throw new NotFoundError(`DFSP for environment ${envId} not found`);
+    }
+    throw error;
+  }
+
+   console.log('dfsp:');
+   console.log(dfsp);
+   if(dfsp.length == 0) {
+     return [];
+   } else {
+
+     let bs = envStatus.filter(phase => phase.phase == PhaseEnum.BUSINESS_SETUP);
+     console.log('bs:');
+     console.log(bs);
+     let s = bs[0].steps.filter(step => step.identifier == StepEnum.ID_GENERATION);
+     console.log('s:');
+     console.log(s);
+
+     if(dfsp[0].id != null && dfsp[0].name != null) {
```

Comment:
not dfsp[0], must be the participant that is requested

Reply:
I think this scenario is not possible. If the DFSP exist, then neither the id nor the name can be null
```
  return knex.schema.createTable('dfsps', (table) => {
    table.increments('id', 11).primary();
    table.integer('env_id', 11).unsigned().notNullable();
    table.string('name', 512).notNullable();
```
one is primary key and the other one is notNullabe

Reply:
Changed per review. Thank you